lib/ai_bot/personas/persona.rb

```diff
@@ -171,6 +172,45 @@ def find_tool(parsed_function)
             persona_options: options[tool_klass].to_h,
           )
         end
+
+        def rag_fragments_prompt(conversation_context)
+          return nil if !SiteSetting.ai_embeddings_enabled?
+          return nil if conversation_context.blank? || uploads.blank?
+
+          latest_interactions =
+            conversation_context
+              .select { |ctx| %i[model user].include?(ctx[:type]) }
+              .map { |ctx| ctx[:content] }
+              .last(10)
+              .join("\n")
+
+          strategy = DiscourseAi::Embeddings::Strategies::Truncation.new
+          vector_rep =
+            DiscourseAi::Embeddings::VectorRepresentations::Base.current_representation(strategy)
+
+          interactions_vector = vector_rep.vector_from(latest_interactions)
+
+          candidate_fragment_ids =
+            vector_rep.asymmetric_rag_fragment_similarity_search(
+              interactions_vector,
+              limit: 10,
```

Comment:
We can take more here, like 50 or 100, and pass to our reranker model and then limit to less.